src/diffusers/image_processor.py

```diff
@@ -0,0 +1,160 @@
+        Convert a numpy image or a batch of images to a PIL image.
+        """
+        if images.ndim == 3:
+            images = images[None, ...]
+        images = (images * 255).round().astype("uint8")
+        if images.shape[-1] == 1:
+            # special case for grayscale (single channel) images
+            pil_images = [Image.fromarray(image.squeeze(), mode="L") for image in images]
+        else:
+            pil_images = [Image.fromarray(image) for image in images]
+
+        return pil_images
+    
+    @staticmethod
+    def numpy_to_pt(images):
+        """
+        Convert a numpy image  to a pytorch tensor
+        """
+        images = torch.from_numpy(images.transpose(0, 3, 1, 2))
+        return images
+
+    @staticmethod
+    def pt_to_numpy(images):
+        """
+        Convert a numpy image  to a pytorch tensor
+        """
+        images = images.cpu().numpy().transpose(0, 2, 3, 1)
+        return images
+
+    @staticmethod
+    def normalize(images):
+        """
+        Normalize an image array to [-1,1]
+        """
+        return 2.0 * images - 1.0
+    
+    def resize(self, images: PIL.Image.Image) -> PIL.Image.Image:
+        """
+        Resize an PIL image. Both height and width will be resized to integer multiple of vae_scale_factor
+        """
+        w, h = images.size
+        w, h = map(lambda x: x - x % self.vae_scale_factor, (w, h))  # resize to integer multiple of vae_scale_factor
+        images = images.resize((w, h), resample=PIL_INTERPOLATION[self.resample])
+        return images
+
+    def encode(
+        self, 
+        image: Union[torch.FloatTensor, PIL.Image.Image, np.ndarray],
+    ) -> torch.Tensor:
+
+        """
+        Preprocess the image input, accpet formats in PIL images, numpy arrays or pytorch tensors"
+        """
+        # convert PIL or list of PIL into numpy 
+        if isinstance(image, PIL.Image.Image):
+            image = [image]
+
+        if isinstance(image[0], PIL.Image.Image):
+            if self.do_resize:
+                image = [self.resize(i) for i in image]
+            image = [np.array(i).astype(np.float32) / 255.0 for i in image]
+
+        if isinstance(image, np.ndarray):
+            image = self.numpy_to_pt(image)
+        elif isinstance(image[0], np.ndarray):
+            image = self.numpy_to_pt(np.stack(image, axis=0))
+        elif not isinstance(image, torch.Tensor) and isinstance(image[0], torch.Tensor):
+            image = torch.cat(image, dim=0)
+        
+        # expected range [0,1], normalize to [-1,1]
+        if image.min() < 0: 
+            warnings.warn(
+                "Passing `image` as torch tensor with value range in [-1,1] is deprecated. The expected value range for image tensor is [0,1] "
+                f"when passing as pytorch tensor or numpy Array. You passed `image` with value range [{image.min()},{image.max()}]",
+                FutureWarning,
+                )
+            self.do_normalize = False
```

Comment:
```suggestion
            do_normalize = False
```

Reply:
Let's not change `self....` attributes depending on a single image. E.g. image one wants to normalize all images and passes first a [-1, 1] image and then a [0, 1] image => this would result in an error